### ISSUE TITLE
Setting up Emails 

### ISSUE THREAD

**Issue by ChrisMaher** [meanjs/mean#1269]
Hi

Just wondering what email provider works with MEAN.js deployed to Heroku.

Gmail blocks the sign in and says its unsecure.

I can find no documentation or suggestion on an email service that works anywhere.

Thanks.

**Comment by mleanos** [comment]
I use Amazon's SES with my MEAN.js app deployed on Heroku. I can't remember if I got Nodemailer working with Heroku.

**Comment by mleanos** [comment]
Actually brainfart.. Obviously, I got Nodemailer working :)

**Comment by ChrisMaher** [comment]
What provider works with it?

**Comment by simison** [comment]
For development [MailDev](https://djfarrelly.github.io/MailDev/) is great.

For production, a friend just collected a list: https://github.com/superlumen/knowledge/wiki/Transactional-Email

@ChrisMaher what's specific about Heroku? Gmail isn't meant for transactional emails so they (Gmail) block if they notice you doing any such activity.

**Comment by ChrisMaher** [comment]
Nothing specific about Heroku apart form that where my App is hosted.

Ill take a look through that list. Any suggestions or are they all straight forward?

**Comment by mleanos** [comment]
Great! Thanks for the list @simison 

I didn't try any of the other providers out. I just went straight to using SES once my app was ready for emails.

**Comment by simison** [comment]
We at @Trustroots were for a year on Mandrill (5-7K emails/mo) but moved to SparkPost last month. All good and easy.

**Comment by ChrisMaher** [comment]
Is there a guide to setting one of these up with Meanjs?

Is it just a matter of getting an email and password from them and modifying this

`mailer: {
    from: process.env.MAILER_FROM || 'email@gmail.com',
    options: {
      service: process.env.MAILER_SERVICE_PROVIDER || 'email',
      auth: {
        user: process.env.MAILER_EMAIL_ID || 'email@gmail.com',
        pass: process.env.MAILER_PASSWORD || 'pass'
      }
    }
  },`

Or do i need to do more than that?

**Comment by mleanos** [comment]
@ChrisMaher Yes. Merely adding your provider authentication settings to this configuration, will do the trick.

**Comment by trendzetter** [comment]
I had to manually swap service to host to use my own smtp server in config/env/development.js
```
-      service: process.env.MAILER_SERVICE_PROVIDER || 'MAILER_SERVICE_PROVIDER',
+      host: process.env.MAILER_HOST || 'out.provider.tld',
```

**Comment by simison** [comment]
@trendzetter `service` is a key for [well known services](https://github.com/nodemailer/nodemailer#using-well-known-services)

List: https://github.com/nodemailer/nodemailer-wellknown#supported-services

https://github.com/meanjs/mean/blob/master/config/env/development.js#L71

Actually configuring Nodemailer at Meanjs would need some better documentation it seems.

**Comment by trendzetter** [comment]
I prefer to use my own server for this purpose.

**Comment by lirantal** [comment]
@codydaig @trendzetter @simison care to update the docs in this regard?

**Comment by mleanos** [comment]
@codydaig @lirantal Can this issue be closed?
